Week3/more_javascript.js

```diff
@@ -0,0 +1,183 @@
+}
+aboutMe(myDetails);
+// I tried also to get this result using for loop 
+// for (let key in myDetails){
+//    console.log(key + ": " +myDetails[key]);
+// }
+
+/* 4-Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike. And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)*/
+
+function vehicleType(color, code){
+  if (code === 1){
+     console.log("a %o car", color )
+   
+   }else if (code ===2)
+   console.log("a %o motorbike", color)
+}
+   vehicleType("blue",1);
+   vehicleType("red",2);
+
+/* 5- Can you write the following without the if statement, but with just as a single line with console.log(...);?
+if (3 === 3) {
+  console.log("yes");
+} else {
+  console.log("no");
+}*/
+
+switch (3===3){
+   case (true):
+      console.log('Yes');
+      break;
+default:
+      console.log('No');
+}
+
+/* 6- Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'*/
+
+function vehicle(color, code, age){
+   if (code === 1 && age > 0 ){
+      console.log("a %o used car", color)
+  } else if (code ===2 && age > 0)
+    console.log("a %o used motorbike", color)
+    else if (code ===1 && age === 0)
+    console.log("a %o new motorbike", color)
+    else if (code ===2 && age === 0)
+    console.log("a %o new motorbike", color)
+}
+vehicle("blue",1,0);
+vehicle("red",1,1);
+
+/* 7- Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.*/
+ 
+let myVehicles = ["motorbike", "caravan", "bike","car"];
+
+/* 8- How do you get the third element from that list?*/
+
+console.log(myVehicles[2]); // third element has index 2
+
+/* 9- Change the function vehicle to use the list of question 7. So thxat vehicle("green", 3, 1) prints "a green new bike".*/
+
+let i=0;
+function vehicles(color,age,i){
+  let index = myVehicles[i];
+  if (age > 0 ){
+      console.log("a %o used %o",color,index)
+  }   
+   else if ( age === 0 )
+      console.log ("a %o new %o",color,index )
+}
+vehicles("green",0,2);
+
+/* 10- Use the list of vehicles to write an advertisement. So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes."*/
+function advertisement (arr) {
+   let last = arr.slice(-1);
+   return arr.slice(0, arr.length - 1).join('s, ') + 's & ' + last+'s';
+}
+console.log("Amazing Joe's Garage, we service "+advertisement(myVehicles));
+
+/*11- What if you add one more vehicle to the list, can you have that added to the advertisement without changing the code for question 10?*/
+myVehicles.push("mopid");
+console.log("Amazing Joe's Garage, we service "+advertisement(myVehicles));
+
+/*12- Create an empty object.*/
+let teachers1 ={};
+let teachers2 = new Object();
+console.log(teachers1);
+console.log(teachers2);
+/* 13- Create an object that contains the teachers that you have had so far for the different modules.*/ 
+let teachers=[{
```

Comment:
This looks nice! But now you've created an array with `objects` inside of it, but I think the assignment is to create an `object` and to have an array inside of it.